src/main/webapp/app/shared/image/secured-image.component.ts

```diff
@@ -1,7 +1,9 @@
-import { Component, Input, OnChanges, EventEmitter, Output } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
-import { BehaviorSubject, Observable } from 'rxjs';
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { BehaviorSubject, isObservable, Observable, of } from 'rxjs';
+import { catchError, filter, map, tap, switchMap } from 'rxjs/operators';
 import { DomSanitizer } from '@angular/platform-browser';
+import { CacheableImageService } from 'app/shared/image/cacheable-image.service';
+import { base64StringToBlob } from 'blob-util';
 
 // Status that is emitted to the client to describe the loading status of the picture
 export const enum QuizEmitStatus {
```

Comment:
I know this has existed before, but I think it's misleading if we use the term Quiz here. Can we rename this to e.g. SecureImageLoadingEmitStatus?

Reply:
Yes, are are right, I renamed it to `ImageLoadingStatus`.